### ISSUE TITLE
Assimp Tutorial

### ISSUE THREAD

**Issue by pdid17** [LWJGL/lwjgl3#302]
The Assimp tutorial included has a few basic bindings, but does not show how to load even a basic file. There are other tutorials on the web, but they are all in a different language, so the bindings are slightly different.

**Comment by Spasi** [comment]
A lot of effort has been put into LWJGL in order to have decent javadoc. There's also extensive documentation and example code on how C concepts map to Java and the LWJGL API. But beyond that, it's outside the scope of LWJGL to teach users how to use the various bindings. There are plenty of demos available (in this repository, in [lwjgl3-demos](https://github.com/LWJGL/lwjgl3-demos), or [elsewhere](https://github.com/LWJGL/lwjgl3-wiki/wiki/2.6.-Tutorial-Index)), but it's not an objective of this project to create demos for every library or feature.

There are many resources one can find to learn a new library and the only issue is how to map C code to Java. There's an objective difficulty associated with that, especially for developers that are only familiar with Java, but it's a necessary step in order to use LWJGL effectively. The concepts involved are generic and can be applied to any binding, but it would be preferable to start with the basics and not dive into Assimp immediately. Assimp does not exactly have a friendly API and there are a lot of structs (which usually cause trouble for new LWJGL users).

If in the above process specific questions arise, they will gladly be answered in the [LWJGL forum](http://forum.lwjgl.org/) or [LWJGL on Slack](https://slack.lwjgl.org/)

With that said, demo contributions are more than welcome!

**Comment by Noviv** [comment]
@Spasi is this still something you'd like done? I recently worked with Assimp a bunch, and I'd love to start contributing to this project!

**Comment by Sylvyrfysh** [comment]
This might get you a starting point

https://github.com/johnnic431/Terminus-Engine/blob/master/src/com/form2bgames/terminusengine/model/AssimpModelWrapper.java

**Comment by Spasi** [comment]
> is this still something you'd like done?

Yes please! Just note that if the demo is relatively big (multiple resources/classes, etc), consider opening a PR in the [lwjgl3-demos](https://github.com/LWJGL/lwjgl3-demos) repository instead of this one.

Btw, the recently added [TootleDemo](https://github.com/LWJGL/lwjgl3/blob/18975883e844d9dc53874836ec45257da13085d9/modules/samples/src/test/java/org/lwjgl/demo/util/tootle/HelloTootle.java) is using Assimp to load simple meshes, [here](https://github.com/LWJGL/lwjgl3/blob/18975883e844d9dc53874836ec45257da13085d9/modules/samples/src/test/java/org/lwjgl/demo/util/tootle/HelloTootle.java#L936). It does geometry processing on load and basic traversing of the mesh data.

**Comment by Noviv** [comment]
Awesome, I'll get to work @Spasi! Small problem though: HelloTootle says my computer does not support GL 3.3. I'm running pretty old integrated hardware (Intel Haswell i915 from more than 5 years ago), but my glxinfo dump still says

```
GLX version: 1.4
    Max core profile version: 3.3
    Max compat profile version: 3.0
    Max GLES1 profile version: 1.1
    Max GLES[23] profile version: 3.1
OpenGL core profile version string: 3.3 (Core Profile) Mesa 17.2.8
OpenGL core profile shading language version string: 3.30
OpenGL version string: 3.0 Mesa 17.2.8
```

which makes me believe I should be able to run it. Not sure what's going on with that (any help would be appreciated), but I'll start with a small Assimp demo!

**Comment by Spasi** [comment]
The demo supports the core profile but uses it on macOS only. Try commenting out the `if` statement at line `114`.

**Comment by Noviv** [comment]
Yep, that worked perfectly. I'll come up with a basic Assimp demo!
